### PR TITLE
feat(privval)!: DO NOT require extension signature

### LINKED ISSUE #2439: privval: DO NOT require extension signature from privval if vote extension is disabled
Refs #2357 

https://github.com/cometbft/cometbft/blob/914e9d7f5e01101198d848be53e7d18e7afacaaa/types/vote.go#L436-L441

Currently, we require an extension signature from the privval EVEN IF vote extensions are disabled. I.e., if the app is not using vote extensions, validators are signing (through HSM or `FilePV`) empty `Vote.Extension` (plus other fields like height and round). Seems like a waste of computing resources. Refs https://github.com/iqlusioninc/tmkms/pull/857#issuecomment-1957368608

### PR DESCRIPTION
**Pull request by melekes**
... from privval if vote extensions are disabled

New API (added sign_extension bool flag):

```
SignVote(chainID string, vote *cmtproto.Vote, signExtension bool) error {
```

New Protobuf message (added sign_extension field):

```
// SignVoteRequest is a request to sign a vote
message SignVoteRequest {
  cometbft.types.v1beta1.Vote vote     = 1;
  string                      chain_id = 2;
  bool sign_extension = 3;
}
```

[api-breaking]
Closes #2439

---

#### PR checklist

- [x] Tests written/updated
- [x] Changelog entry added in `.changelog` (we use [unclog](https://github.com/informalsystems/unclog) to manage our changelog)
- [ ] Updated relevant documentation (`docs/` or `spec/`) and code comments
- [x] Title follows the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/) spec

### CODE REVIEW

**Comment by melekes** [comment]
This PR won't be backported to `v0.38.x` because it's API-breaking.

File: .changelog/unreleased/improvements/2496-privval-ext-signature.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+- `[privval]` DO NOT require extension signature from privval if vote extension
```
Comment (cason) [inline .changelog/unreleased/improvements/2496-privval-ext-signature.md:1]:
```suggestion
- `[privval]` DO NOT require extension signature from privval if vote extensions
```

File: .changelog/unreleased/improvements/2496-privval-ext-signature.md [hunk 2]
```diff
@@ -0,0 +1,5 @@
...
+  is disabled [\#2496](https://github.com/cometbft/cometbft/pull/2496)
```
Comment (cason) [inline .changelog/unreleased/improvements/2496-privval-ext-signature.md:2]:
```suggestion
  are disabled [\#2496](https://github.com/cometbft/cometbft/pull/2496)
```

File: privval/signer_client_test.go [hunk 3]
```diff
@@ -433,7 +441,54 @@ func TestSignerUnexpectedResponse(t *testing.T) {
...
 
-		e := tc.signerClient.SignVote(tc.chainID, want.ToProto())
+		e := tc.signerClient.SignVote(tc.chainID, want.ToProto(), false)
 		require.ErrorIs(t, e, cmterrors.ErrRequiredField{Field: "response"})
 	}
 }
+
+func TestSignerVoteExtension(t *testing.T) {
+	for _, tc := range getSignerTestCases(t) {
+		ts := time.Now()
```
Comment (cason) [inline privval/signer_client_test.go:451]: Use `cmttime`.

**Review by cason — APPROVED** [review]
Legit.

**Review by lasarojc — COMMENTED** [review]
Could it be made so the lack of the extra field results in the current behavior being followed, so that it could be backported to the 0.38?

**Comment by melekes** [comment]
> Could it be made so the lack of the extra field results in the current behavior being followed, so that it could be backported to the 0.38?

I don't see a need given the time it will take to come up with some intermediate solution. All signers currently follow the standard behavior (sign all the time). Shipping as part of v1 is okay imho.

**Comment by cason** [comment]
We probably need to backport this, given: https://github.com/cometbft/cometbft/issues/2711

**Comment by melekes** [comment]
> We probably need to backport this, given: #2711

https://github.com/cometbft/cometbft/pull/2496#issuecomment-1974753148

**Comment by sergio-mena** [comment]
Agree with @melekes ... if absolutely needed, we could backport to experimental branch

**Pull request merged**